### PR TITLE
Add LockSupport TaskBlock profiling instrumentation

### PR DESCRIPTION
**Pull request by kaahos**
# What Does This Do

Add TaskBlock profiling instrumentation for `LockSupport.park/unpark`, including blocker and unblocking span attribution where available.

# Motivation

`LockSupport.park` is widely used by executors, queues, schedulers, and virtual-thread internals. Capturing these park intervals as TaskBlock events makes parked time attributable to spans and easier to separate from active execution in wall-clock profiles.

# Additional Notes

- Stacked on https://github.com/DataDog/dd-trace-java/pull/11544
- Add `DD_TRACE_LOCK_SUPPORT_ENABLED`

# Contributor Checklist

- Format the title according to [the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any other useful labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- Avoid using `close`, `fix`, or [any linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- Update the [CODEOWNERS](https://github.com/DataDog/dd-trace-java/blob/master/.github/CODEOWNERS) file on source file addition, migration, or deletion
- Update [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) with any new configuration flags or behaviors
- Add your completed PR to the merge queue by commenting `/merge`. You can also:
  - Customize the commit message associated with the merge with `/merge --commit-message "..."`
  - Remove your PR from the merge queue with `/merge -c`
  - Skip all merge queue checks with `/merge -f --reason "reason"`; please use this judiciously, as some checks do not run at the PR-level
  - Get more information in [this doc](https://datadoghq.atlassian.net/wiki/spaces/DEVX/pages/3121612126/MergeQueue)

Jira ticket: [[PROF-15321]](https://datadoghq.atlassian.net/browse/PROF-15321)

[PROF-15321]: https://datadoghq.atlassian.net/browse/PROF-15321?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java [hunk 1]
```diff
@@ -119,6 +120,14 @@ public final class ProfilingConfig {
       "profiling.ddprof.wall.context.filter";
   public static final boolean PROFILING_DATADOG_PROFILER_WALL_CONTEXT_FILTER_DEFAULT = true;
 
+  public static final String PROFILING_DATADOG_PROFILER_WALL_PRECHECK =
+      "profiling.ddprof.wall.precheck";
+  public static final boolean PROFILING_DATADOG_PROFILER_WALL_PRECHECK_DEFAULT = false;
+
+  public static final String PROFILING_TASK_BLOCK_UNBLOCKING_ENABLED =
```
Comment (jbachorik) [inline dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java]: **[Sphinx Review — MEDIUM]** `PROFILING_TASK_BLOCK_UNBLOCKING_ENABLED` and its `_DEFAULT` are defined but never read anywhere in the codebase. The unblocking-span attribution path (`recordUnpark` writing `UNPARKING_SPAN`) runs unconditionally with no kill switch. The constant is also absent from `supported-configurations.json`.

**Suggestion:** Either wire the flag into `LockSupportHelper.recordUnpark()` (guard `UNPARKING_SPAN` population when disabled) and register `DD_PROFILING_TASKBLOCK_UNBLOCKING_ENABLED` in `supported-configurations.json`, or delete both constants.

*[CONSENSUS · concern: necessity · confidence: high]*

File: metadata/supported-configurations.json [hunk 2]
```diff
@@ -2753,6 +2753,14 @@
         "aliases": []
       }
     ],
+    "DD_PROFILING_ASYNC_WALL_PRECHECK": [
```
Comment (jbachorik) [inline metadata/supported-configurations.json]: **[Sphinx Review — MEDIUM]** Spurious metadata entry `DD_PROFILING_ASYNC_WALL_PRECHECK`: no `profiling.async.wall.precheck` property exists in the codebase. The only precheck config is `profiling.ddprof.wall.precheck`, which maps to `DD_PROFILING_DDPROF_WALL_PRECHECK` and is already present at line 2996. This phantom entry will be surfaced to docs generators and Helm chart consumers as a real env-var.

**Suggestion:** Remove the `DD_PROFILING_ASYNC_WALL_PRECHECK` block. The precheck flag is already exposed via `DD_PROFILING_DDPROF_WALL_PRECHECK`.

*[concern: consistency · confidence: high]*

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    if (ctx == null) {
+      UNPARKING_SPAN.remove(Thread.currentThread());
+      return null;
+    }
+    try {
+      profiling.parkEnter();
+    } catch (Throwable ignored) {
+      // parkEnter failed (e.g. profiler not yet initialised, JNI error); do not track this park.
+      // Drain any stale unblocking-span entry so it is not mis-attributed to the next park.
+      UNPARKING_SPAN.remove(Thread.currentThread());
```
Comment (jbachorik) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java]: **[Sphinx Review — LOW]** `recordUnpark` inserts an entry for the target thread on every traced unpark, but draining only happens inside an instrumented park return. The unpark-before-park idiom (a normal `LockSupport` pattern where the permit is set before the thread parks) leaves a stale entry that the next unrelated instrumented park on that thread will consume and mis-attribute as being unblocked by the wrong span.

**Suggestion:** Extend the existing 'Known limitation' javadoc to explicitly cover the unpark-before-park staleness window. For stronger correctness, stamp entries with a monotonic sequence and ignore stale entries on park return.

*[concern: correctness · confidence: medium]*

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+              unblockingSpanId,
+              state.spanId,
+              state.rootSpanId);
+        } catch (Throwable ignored) {
+        }
+      }
+    } else {
+      // Platform thread: parkExit() clears native parked state and records an eligible TaskBlock
+      // using the entry tick saved by parkEnter().
+      state.profiling.parkExit(state.blockerHash, unblockingSpanId);
```
Comment (jbachorik) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java]: **[Sphinx Review — LOW]** Asymmetric exception handling in `finish(ParkState,long)`: the virtual-thread branch wraps `recordTaskBlockWithContext` in `try/catch(Throwable)` but the platform-thread branch calls `parkExit()` unguarded. The advice `suppress=Throwable` covers the production call path, but the inconsistency is a latent reliability hole for direct callers.

**Suggestion:** Wrap the `parkExit()` call in `try { ... } catch (Throwable ignored) {}` to match the virtual-thread branch.

*[concern: robustness · confidence: medium]*

File: dd-java-agent/instrumentation/datadog/profiling/lock-support/src/test/java/datadog/trace/instrumentation/locksupport/LockSupportProfilingInstrumentationTest.java [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+  /**
+   * Regression test for stale-entry misattribution.
+   *
+   * <p>If {@code unpark(t)} is called (inserting an entry into {@code UNPARKING_SPAN}) and thread
+   * {@code t} then parks without an active span ({@code state == null}), the entry must still be
+   * drained. Without the fix, it would linger and be incorrectly attributed to the next {@code
+   * TaskBlock} emitted on that thread.
+   */
+  @Test
+  void stale_entry_is_drained_when_park_fires_without_active_span() {
```
Comment (jbachorik) [inline dd-java-agent/instrumentation/datadog/profiling/lock-support/src/test/java/datadog/trace/instrumentation/locksupport/LockSupportProfilingInstrumentationTest.java]: **[Sphinx Review — LOW]** Drain-semantics coverage gap. The one-arg `finish(ParkState)` overload drains `UNPARKING_SPAN` before delegating; the public two-arg `finish(ParkState,long)` with `null` state returns early without touching the map. Tests exercise only the one-arg drain path.

**Suggestion:** Either add a test asserting `finish(null, x)` does NOT drain (documenting the intended asymmetry), or make the two-arg overload also drain on null state for uniform behavior.

*[concern: completeness · confidence: medium]*

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+      return null;
+    }
+    long blockerHash = blocker != null ? System.identityHashCode(blocker) : 0L;
+    if (VirtualThreads.isCurrent()) {
+      // Virtual thread: skip native parkEnter0 (carrier-scoped TLS is unsafe).
+      // Capture span/root ids now so we can pass them explicitly on unpark.
+      ProfilerContext ctx = ProfilerContexts.of(AgentTracer.activeSpan());
+      if (ctx == null) {
+        // No active span - nothing to record.
+        UNPARKING_SPAN.remove(Thread.currentThread());
```
Comment (jbachorik) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java]: **[Sphinx Review — LOW]** The virtual-thread branch of `captureState` (`if (VirtualThreads.isCurrent())`) is never executed in unit tests — tests always run on platform threads. Inverting the condition would pass all tests silently. Span/root ID capture, `getCurrentTicks()`, and `recordTaskBlockWithContext` invocation have no deterministic unit coverage.

**Suggestion:** Add a unit test that constructs a virtual `ParkState` directly (5-arg constructor) and calls `finish`, asserting `recordTaskBlockWithContext` is invoked and that the `spanId==0` guard suppresses emission when inactive.

*[concern: test-adequacy · confidence: high · mutation-detected]*

File: dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java (hunk 1 above)
Comment (kaahos) [reply dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java]: addressed in 84fded4bf9.

File: metadata/supported-configurations.json (hunk 2 above)
Comment (kaahos) [reply metadata/supported-configurations.json]: addressed in 6ed3220ae0.

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java (hunk 3 above)
Comment (kaahos) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java]: addressed in a20d06dcf2.

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java (hunk 4 above)
Comment (kaahos) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java]: addressed in a20d06dcf2.

File: dd-java-agent/instrumentation/datadog/profiling/lock-support/src/test/java/datadog/trace/instrumentation/locksupport/LockSupportProfilingInstrumentationTest.java (hunk 5 above)
Comment (kaahos) [reply dd-java-agent/instrumentation/datadog/profiling/lock-support/src/test/java/datadog/trace/instrumentation/locksupport/LockSupportProfilingInstrumentationTest.java]: addressed in a20d06dcf2.

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java (hunk 6 above)
Comment (kaahos) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/LockSupportHelper.java]: addressed in a20d06dcf2.
